File: analytics/src/main/java/io/appmetrica/analytics/impl/billing/AutoInappCollectingInfoSerializer.java
package io.appmetrica.analytics.impl.billing;

import androidx.annotation.NonNull;
import io.appmetrica.analytics.impl.db.protobuf.BaseProtobufStateSerializer;
import io.appmetrica.analytics.impl.protobuf.client.AutoInappCollectingInfoProto;
import java.io.IOException;

public class AutoInappCollectingInfoSerializer extends
        BaseProtobufStateSerializer<AutoInappCollectingInfoProto.AutoInappCollectingInfo> {

    @NonNull
    @Override
    public AutoInappCollectingInfoProto.AutoInappCollectingInfo toState(@NonNull byte[] data) throws IOException {
        return AutoInappCollectingInfoProto.AutoInappCollectingInfo.parseFrom(data);
    }

    @NonNull
    @Override
    public AutoInappCollectingInfoProto.AutoInappCollectingInfo defaultValue() {
        return new AutoInappCollectingInfoProto.AutoInappCollectingInfo();
    }
}

File: billing/src/main/java/io/appmetrica/analytics/billing/impl/storage/AutoInappCollectingInfoSerializer.kt
package io.appmetrica.analytics.billing.impl.storage

import io.appmetrica.analytics.billing.impl.protobuf.client.AutoInappCollectingInfoProto
import io.appmetrica.analytics.coreutils.internal.data.BaseProtobufStateSerializer

class AutoInappCollectingInfoSerializer :
    BaseProtobufStateSerializer<AutoInappCollectingInfoProto.AutoInappCollectingInfo>() {

    override fun toState(data: ByteArray): AutoInappCollectingInfoProto.AutoInappCollectingInfo {
        return AutoInappCollectingInfoProto.AutoInappCollectingInfo.parseFrom(data)
    }

    override fun defaultValue(): AutoInappCollectingInfoProto.AutoInappCollectingInfo {
        return AutoInappCollectingInfoProto.AutoInappCollectingInfo()
    }
}
